pyproject.toml

```diff
@@ -0,0 +1,144 @@
+classifiers = [
+    #   3 - Alpha
+    #   4 - Beta
+    #   5 - Production/Stable
+    "Development Status :: 4 - Beta",
+    "Intended Audience :: Science/Research",
+    "Topic :: Scientific/Engineering :: Artificial Intelligence",
+    "Topic :: Software Development :: Libraries :: Python Modules",
+    "License :: OSI Approved :: MIT License",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+]
+exclude = ["test/*", "examples/*", "docs/*"]
+
+[tool.poetry.dependencies]
+python = "^3.9"
+gymnasium = ">=0.29.0"
+h5py = ">=3.9.0"
+numba = ">=0.57.1"
+numpy = ">=1"
+packaging = "*"
+pettingzoo = ">=1.22"
+ray = ">=2"
+tensorboard = ">=2.5.0"
+# Torch 2.0.1 causes problems, see https://github.com/pytorch/pytorch/issues/100974
+torch = "==2.0.0"
+tqdm = "*"
+
+# TODO: add versions
+atari_py = {version = "*", optional = true}
+envpool = {version = "^0.8.2", optional = true}
+mujoco_py = {version = "*", optional = true}
+opencv_python = {version = "*", optional = true}
+pybullet = {version = "*", optional = true}
+
+[tool.poetry.extras]
+atari = ["atari_py", "opencv-python"]
+mujoco = ["mujoco_py"]
+pybullet = ["pybullet"]
+envpool = ["envpool"]
+
+
+[tool.poetry.group.dev]
+optional = true
+[tool.poetry.group.dev.dependencies]
+black = "*"
+doc8 = "*"
+jinja2 = "*"
+mypy = ">=1.4.1"
+# networkx is used in a test
+networkx = "*"
+poethepoet = ">=0.20.0"
+pre-commit = ">=3.3.3"
+pygame = ">=2.1.0"
+pymunk = ">=6.2.1"
+pytest = "*"
+pytest-cov = "*"
+ruff = "*"
+scipy = "*"
+shimmy = "*"
+sphinx = "<7"
+sphinx_rtd_theme = "*"
+sphinxcontrib-bibtex = "*"
+wandb = ">=0.12.0"
+
+[tool.black]
+line-length = 88
+target-version = ["py39"]
+
+[tool.mypy]
+files = "tianshou/**/*.py"
+allow_redefinition = true
+check_untyped_defs = true
+disallow_incomplete_defs = true
+disallow_untyped_defs = true
+ignore_missing_imports = true
+no_implicit_optional = true
+pretty = true
+show_error_codes = true
+show_error_context = true
+show_traceback = true
+strict_equality = true
+strict_optional = true
+warn_no_return = true
+warn_redundant_casts = true
+warn_unreachable = true
+warn_unused_configs = true
+warn_unused_ignores = true
+
+[tool.doc8]
+max-line-length = 1000
+
+[tool.ruff]
+fix = true
+ignore-init-module-imports = true
+line-length = 88
+select = ["A", "ASYNC", "B", "C4", "C90", "D", "DTZ", "E", "F", "FLY", "I", "ISC", "N", "NPY", "PGH", "PIE", "PLC", "PLE", "PLR", "PLW", "PT", "RET", "RUF", "RSE", "SIM", "TID", "UP", "W", "YTT"]
+ignore = ["E501", "PGH001", "PGH002", "PGH003", "RET504", "S101"]
+unfixable = ["F401", "F841"]
+src = ["tianshou", "test"]
+target-version = "py39"
+
+[tool.ruff.pydocstyle]
+ignore-decorators = ["D100" ,"D102", "D104", "D105", "D107", "D203", "D213", "D401", "D402"]
+# exclude = "^(test/)|(docs/)|(examples/)|(setup.py)"
+
+[tool.poetry_bumpversion.file."tianshou/__init__.py"]
+
+[tool.poetry-sort]
```

Comment:
is there a command for package building? Similar to `python setup.py sdist bdist_wheel`

Reply:
yeah, `poetry build` should do the job. You can also publish with `poetry publish`. Let's maybe discuss the release process at some point, as mentioned in a thread above